causalflows/distributions.py

```diff
@@ -2,36 +2,36 @@
 
 __all__ = ['CausalNormalizingFlow']
 
-import torch
-
+from collections.abc import Generator, Sequence
 from contextlib import contextmanager
+from typing import Any, Self, cast
+
+import torch
 from torch import LongTensor, Size, Tensor
 from torch.distributions import Distribution, Transform
-from typing import List
 from zuko.distributions import NormalizingFlow
 
 empty_size = Size()
 
 
 class IntervenedTransform(Transform):
-    def __init__(self, transform: Transform, index: List[LongTensor], value: List[Tensor]):
+    def __init__(self, transform: Transform, index: list[LongTensor], value: list[Tensor]) -> None:
         super().__init__()
         self.transform = transform
-        self.index: List[LongTensor] = index
-        self.value: List[Tensor] = value
-
-    def _inv_call(self, u):
-        index = torch.cat(self.index, dim=-1)
-        value = torch.stack(self.value, dim=-1)
-
-        x = self.transform.inv(u)
-        x[..., index] = value.to(device=x.device)
-        u_tmp = self.transform(x)
-        u[..., index] = u_tmp[..., index]
+        self.index = cast(
+            LongTensor, torch.cat(tuple(index), dim=-1)
+        )  # tuple is must for torch.cat typing
```

Comment:
what version of pytorch are you using? I have Pytorch 2.6 and I get `tensors: tuple[Tensor, ...] | list[Tensor] | None` as the type for cat argument

Reply:
yes that is what pytorch annotate, but due to variance of the containers only `tuple` can be used here (or another cast) to make it pass type-checking.
(alternatively we just allow a type-checking error to make code concise. up to you)